swap_meet/vendor.py

```diff
@@ -1,2 +1,87 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        
+        return False
+    
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            # we do not need to import id bc we're not inheriting anything
+            if item.id == item_id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        other_vendor reps the friend that the vendor(self) is swapping with
+        my_item reps the item I am giving away
+        their_item reps item other_vendor plans to give
+        '''
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        # remove my_item from inv
+        self.remove(my_item)
+        # add my_item to other_vendor
+        other_vendor.add(my_item)
+        # remove their_item from the other Vendors inv
+        other_vendor.remove(their_item)
+        # add their_item to my inv
+        self.add(their_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+        self.swap_items(other_vendor,my_first_item, their_first_item)
```

Comment:
I recommend returning what `swap_items` returns for these, just in case it changes in the future in some way that makes it return `False`.

```python
        return self.swap_items(other_vendor,my_first_item, their_first_item)
```